### PR TITLE
apache camel propagation tests - cleanup #1

### PR DESCRIPTION
**Pull request by kubawach**
Clean-up after finishing AWS / Camel propagation tests.

This is the first part to show the direction of changes - once approved I'll follow up with the rest of tests (and perhaps also AWS SDK ones).

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy [hunk 1]
```diff
@@ -185,195 +111,259 @@ class S3CamelTest extends AgentInstrumentationSpecification {
...
             "net.transport" "IP.TCP"
           }
         }
       }
-      // camel polling
-      trace(4, 1) {
+      trace(2, 1) {
         span(0) {
-          name "SQS.ReceiveMessage"
+          name "SQS.DeleteMessage"
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy:121]: Can you use `AwsSpan.sqs()` instead?

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy [hunk 2]
```diff
@@ -185,195 +111,259 @@ class S3CamelTest extends AgentInstrumentationSpecification {
...
             "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
           }
         }
       }
-      trace(1, 1) {
+      trace(4, 1) {
         span(0) {
-          name "S3.CreateBucket"
+          name "SNS.Subscribe"
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy:201]: Hmm, maybe for all SNS spans you could add a new `AwsSpan.sns()` method?

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpan.groovy [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        "http.status_code" 200
+        "http.url" String
+        "net.peer.name" String
+        "net.transport" "IP.TCP"
+        "net.peer.port" { it == null || Number }
+      }
+    }
+  }
+
+  static sqs(TraceAssert traceAssert, int index, spanName, queueUrl=null, queueName=null, spanKind=CLIENT, parentSpan=null) {
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpan.groovy:40]: 👍 👍 👍

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy (hunk 1 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy:121]: Yep, that's the idea of a follow-up once I'm sure these changes are heading in the right direction.

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpan.groovy (hunk 3 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpan.groovy:40]: :D

File: instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy (hunk 2 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy:201]: Yep, that's the idea of a follow-up once I'm sure these changes are heading in the right direction.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**